apps/src/code-studio/pd/foorm/Foorm.jsx

```diff
@@ -58,26 +69,84 @@ export default class Foorm extends React.Component {
     if (this.props.submitParams) {
       requestData = {...this.props.submitParams, ...requestData};
     }
+    this.setState({
+      statusMessage: 'Your responses are being submitted...',
+      hasError: false,
+      submitting: true
+    });
+    const startTime = Date.now();
     $.ajax({
       url: this.props.submitApi,
       type: 'post',
       dataType: 'json',
       data: requestData
-    });
+    })
+      .done(() => {
+        this.setCompletedStatusMessage(
+          startTime,
+          'Thank you for completing the survey! Your responses saved successfully.',
+          false,
+          null
+        );
+      })
+      .fail(jqXHR => {
+        if (jqXHR.status === 409) {
+          this.setCompletedStatusMessage(
+            startTime,
+            'You already submitted a response. Thank you!',
+            false,
+            null
+          );
+        } else {
+          this.setCompletedStatusMessage(
+            startTime,
+            'An error occurred and we could not save your response.',
+            true,
+            survey
+          );
+        }
+      });
   };
 
+  setCompletedStatusMessage(startTime, statusMessage, hasError, survey) {
+    const elapsedTime = Date.now() - startTime;
+    // if the api request took less than 1 second, wait until 1 second has
+    // passed so spinner does not flash on the screen.
+    const timeout =
+      elapsedTime > SPINNER_WAIT_MS ? 0 : SPINNER_WAIT_MS - elapsedTime;
```

Comment:
An alternate might be `timeout = Math.max(0, SPINNER_WAIT_MS - elapsedTime)` but I think yours is more readable.

Reply:
You might also make yours check `elapsedTime >= SPINNER_WAIT_MS` since in that case you'll hit a 0 in the second case anyway, so might as well make it explicit.

Reply:
One more thing: when running this locally, the spinner seems to show for closer to 3 seconds.  I think that's probably more natural-feeling than 1 second, but any idea why it's up for that long?

Reply:
Turns out I just had a long network event.  As discussed, 2 seconds is probably a good minimum.

Reply:
updated to 2 seconds and changed to >=